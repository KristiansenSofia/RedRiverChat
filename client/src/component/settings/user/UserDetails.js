import React, { Component } from 'react';

// Import NPM-modules
import Typography from 'material-ui/Typography';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import axios from 'axios';

// Import styles. settingsUserStyles for all imported components with a style attribute and CSS-file for classNames and id.
import {settingsUserStyles} from "../../../styles/SettingsStyles";
import '../../../styles/Styles.css'
import {registerStyles} from "../../../styles/AuthStyles";

/**
 *  UserInfo-component.
 *
 *  @author Jimmy
 */

class SettingsUser extends Component {

    constructor(props){
        super(props);

        this.state = {
            userName: this.props.state.userInfo.username,
            firstName: this.props.state.userInfo.firstName,
            surname: this.props.state.userInfo.surname,
            email: this.props.state.userInfo.email,
            streetAddress: this.props.state.userInfo.streetAddress,
            zipCode: this.props.state.userInfo.postcode,
            city: this.props.state.userInfo.city,
            socialSecurity: this.props.state.userInfo.socialSecurity,
            telephoneNumber: this.props.state.userInfo.telephoneNumber,
            relativeUsername: this.props.state.userInfo.relativeUsername,
            editDetails: false,
        };

    }

    renderForm() {

        return (
            <form style={settingsUserStyles.container} noValidate autoComplete="off">
                <TextField
                    id="userName"
                    label="Användarnamn"
                    style={settingsUserStyles.textField}
                    value={this.state.userName}
                    onChange={this.handleChange('userName')}
                    margin="normal"
                />
                <TextField
                    id="email"
                    label="Email"
                    style={settingsUserStyles.textField}
                    value={this.state.email}
                    onChange={this.handleChange('email')}
                    margin="normal"
                    type="email"
                />
                <TextField
                    id="firstName"
                    label="Förnamn"
                    style={settingsUserStyles.textField}
                    value={this.state.firstName}
                    onChange={this.handleChange('firstName')}
                    margin="normal"
                />
                <TextField
                    id="surname"
                    label="Efternamn"
                    style={settingsUserStyles.textField}
                    value={this.state.surname}
                    onChange={this.handleChange('surname')}
                    margin="normal"
                />
                <TextField
                    id="streetAddress"
                    label="Adress"
                    style={settingsUserStyles.textField}
                    value={this.state.streetAddress}
                    onChange={this.handleChange('streetAddress')}
                    margin="normal"
                />
                <TextField
                    id="zipCode"
                    label="Postnummer"
                    style={settingsUserStyles.textField}
                    value={this.state.zipCode}
                    onChange={this.handleChange('zipCode')}
                    margin="normal"
                    type="number"
                />
                <TextField
                    id="city"
                    label="Postort"
                    style={settingsUserStyles.textField}
                    value={this.state.city}
                    onChange={this.handleChange('city')}
                    margin="normal"
                />
                <TextField
                    id="socialSecurity"
                    label="Personnummer"
                    style={settingsUserStyles.textField}
                    value={this.state.socialSecurity}
                    onChange={this.handleChange('socialSecurity')}
                    margin="normal"
                />
                <TextField
                    id="telephoneNumber"
                    label="Telefonnummer"
                    style={settingsUserStyles.textField}
                    value={this.state.telephoneNumber}
                    onChange={this.handleChange('telephoneNumber')}
                    margin="normal"
                    type="tel"
                />
                <TextField
                    id="relativeUsername"
                    label="Anhörigs namn"
                    style={settingsUserStyles.textField}
                    value={this.state.relativeUsername}
                    onChange={this.handleChange('relativeUsername')}
                    margin="normal"
                />
                <div className="RegisterButton">
                    <Button variant="raised" style={settingsUserStyles.button} onClick={this.handleSubmit}>
                        Registrera
                    </Button>
                </div>
            </form>
        );
    }

    render() {
        return (
            <div className="UserDetails">

            </div>
        );
    }
}

export default SettingsUser;