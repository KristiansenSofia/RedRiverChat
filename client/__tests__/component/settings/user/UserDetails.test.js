import React from 'react';
import Enzyme from 'enzyme';
import {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UserDetails from '../../../../src/component/settings/user/UserDetails';

Enzyme.configure({ adapter: new Adapter() });

test('UserDetails renders without crashing', () => {
  const component = shallow(
    <UserDetails state =  {{ userInfo : { username: 'username' } }}/>
  );

  expect(component).toMatchSnapshot();
});


test('UserDetails handleChange should set userName to Banan', () => {
  const component = shallow(
    <UserDetails state =  {{ userInfo : { username: 'username' } }} />
  );

  let event = {
    target: {
      value: 'Banan'
    }
  }

  component.instance().handleChange('userName')(event);
  const userState = component.state('userName');

  expect(userState).toEqual('Banan');
  expect(userState).not.toEqual('username');
});


test('UserDetails handleChange should set firstName to Jan', () => {
  const component = shallow(
    <UserDetails state =  {{ userInfo : { firstName: 'firstname' } }} />
  );

  let event = {
    target: {
      value: 'Jan'
    }
  }

  component.instance().handleChange('firstName')(event);
  const firstNameState = component.state('firstName');

  expect(firstNameState).toEqual('Jan');
  expect(firstNameState).not.toEqual('firstname');
});


test('UserDetails handleEditButton should set formDisabled to false', () => {
  const component = shallow(
    <UserDetails state =  {{ userInfo : { firstName: 'firstname' } }} />
  );

  component.instance().handleEditButton(false);
  const formState = component.state('formDisabled');

  expect(formState).toEqual(false);
  expect(formState).not.toEqual(true);
});


test('UserDetails handleEditButton should set formDisabled to true', () => {
  const component = shallow(
    <UserDetails state =  {{ userInfo : { firstName: 'firstname' } }} />
  );

  component.instance().handleEditButton(true);
  const formState = component.state('formDisabled');

  expect(formState).toEqual(true);
  expect(formState).not.toEqual(false);
});


test('UserDetails handleSubmit should return "Formuläret ej korrekt ifyllt!" when username is missing', () => {
  const baseProps = {
    openSnackBar: jest.fn(),
    state: {
      userInfo: {
        firstName: 'firstname'
      }
    }
  };

  const component = shallow(
    <UserDetails {...baseProps} />
  );

  component.setState({
    userName: '',
    email: 'banan@example.com',
    surname: 'Banan',
    firstName: 'Jan'
  });

  component.instance().handleSubmit();

  expect(baseProps.openSnackBar).toHaveBeenCalledTimes(1);
});


test('UserDetails handleSubmit should return "Formuläret ej korrekt ifyllt!" when email is missing', () => {
  const baseProps = {
    openSnackBar: jest.fn(),
    state: {
      userInfo: {
        firstName: 'firstname'
      }
    }
  };

  const component = shallow(
    <UserDetails {...baseProps} />
  );

  component.setState({
    userName: 'Banan',
    email: '',
    surname: 'Banan',
    firstName: 'Jan'
  });

  component.instance().handleSubmit();

  expect(baseProps.openSnackBar).toHaveBeenCalledTimes(1);
});


test('UserDetails handleSubmit should return "Formuläret ej korrekt ifyllt!" when surname is missing', () => {
  const baseProps = {
    openSnackBar: jest.fn(),
    state: {
      userInfo: {
        firstName: 'firstname'
      }
    }
  };

  const component = shallow(
    <UserDetails {...baseProps} />
  );

  component.setState({
    userName: 'Banan',
    email: 'banan@example.com',
    surname: '',
    firstName: 'Jan'
  });

  component.instance().handleSubmit();

  expect(baseProps.openSnackBar).toHaveBeenCalledTimes(1);
});


test('UserDetails handleSubmit should return "Formuläret ej korrekt ifyllt!" when surname is missing', () => {
  const baseProps = {
    openSnackBar: jest.fn(),
    state: {
      userInfo: {
        surname: 'surname'
      }
    }
  };

  const component = shallow(
    <UserDetails {...baseProps} />
  );

  component.setState({
    userName: 'Banan',
    email: 'banan@example.com',
    surname: 'Banan',
    firstName: ''
  });

  component.instance().handleSubmit();

  expect(baseProps.openSnackBar).toHaveBeenCalledTimes(1);
});


test('UserDetails handleSubmit should return "Lösenorden matchar inte!" when password and passwordConfirm is not matching', () => {
  const baseProps = {
    openSnackBar: jest.fn(),
    state: {
      userInfo: {
        surname: 'surname'
      }
    }
  };

  const component = shallow(
    <UserDetails {...baseProps} />
  );

  component.setState({
    userName: 'Banan',
    password: 'Password123',
    email: 'banan@example.com',
    passwordConfirm: 'password1',
    surname: 'Banan',
    firstName: 'Jan'
  });

  component.instance().handleSubmit();

  const passwordState = component.state('password');
  const passwordConfirmState = component.state('passwordConfirm');

  expect(baseProps.openSnackBar).toHaveBeenCalledTimes(1);
  expect(passwordState).toEqual('Password123');
  expect(passwordConfirmState).toEqual('password1');
  expect(passwordState).not.toEqual(passwordConfirmState);
});


test('UserDetails handleSubmit should call handleEditButton', () => {
  const baseProps = {
    openSnackBar: jest.fn(),
    state: {
      userInfo: {
        surname: 'surname'
      }
    }
  };

  const component = shallow(
    <UserDetails {...baseProps} />
  );

  component.setState({
    userName: 'Banan',
    password: 'Password123',
    email: 'banan@example.com',
    passwordConfirm: 'Password123',
    surname: 'Banan',
    firstName: 'Jan'
  });

  component.instance().handleEditButton = jest.fn();

  component.instance().handleSubmit();

  expect(baseProps.openSnackBar).toHaveBeenCalledTimes(0);
  expect(component.instance().handleEditButton).toHaveBeenCalledTimes(1);
});


test('UserDetails render should call handleEditButton when "Ändra uppgifter" button is clicked', () => {
  const baseProps = {
    state: {
      userInfo: {
        surname: 'surname'
      }
    }
  };

  const component = mount(
    <UserDetails {...baseProps} />
  );

  component.setState({
    formDisabled: true
  });

  component.instance().handleEditButton = jest.fn();

  let button = component.find('button');

  component.instance().render();
  button.simulate('click');

  expect(component.instance().handleEditButton).toHaveBeenCalledTimes(1);
});


test('UserDetails render should call handleEditButton when "Ångra" button is clicked', () => {
  const baseProps = {
    state: {
      userInfo: {
        surname: 'surname'
      }
    }
  };

  const component = mount(
    <UserDetails {...baseProps} />
  );

  component.setState({
    formDisabled: false
  });

  component.instance().handleEditButton = jest.fn();

  let button = component.find('button');

  component.instance().render();
  button.at(0).simulate('click');

  expect(component.instance().handleEditButton).toHaveBeenCalledTimes(1);
});


test('UserDetails render should call handleSubmit when "Ok" button is clicked', () => {
  const baseProps = {
    state: {
      userInfo: {
        surname: 'surname'
      }
    }
  };

  const component = mount(
    <UserDetails {...baseProps} />
  );

  component.setState({
    formDisabled: false
  });

  component.instance().handleSubmit = jest.fn();

  let button = component.find('button');

  component.instance().render();
  button.at(1).simulate('click');

  expect(component.instance().handleSubmit).toHaveBeenCalledTimes(1);
});
