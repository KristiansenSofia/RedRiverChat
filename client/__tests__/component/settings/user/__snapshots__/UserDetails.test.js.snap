// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails renders without crashing 1`] = `
<div
  className="UserDetails"
>
  <h1
    className="MuiTypography-root-1 MuiTypography-headline-6 MuiTypography-alignLeft-13"
    style={
      Object {
        "fontSize": "90%",
        "marginLeft": 30,
      }
    }
  >
    Anv√§ndaruppgifter
  </h1>
</div>
`;
